File: 0036-valid-sudoku/0036-valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        rows = [[0] * 9 for _ in range(9)]
        cols = [[0] * 9 for _ in range(9)]
        boxes = [[0] * 9 for _ in range(9)]

        for i in range(9):
            for j in range(9):
                if board[i][j] == '.':
                    continue
                num = int(board[i][j]) -1 
                boxId = (i // 3) * 3 + j // 3
                if rows[i][num] or cols[j][num] or boxes[boxId][num]:
                    return False
                rows[i][num] = 1
                cols[j][num] = 1
                boxes[boxId][num] = 1
        return True

File: 0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function (board) {

    const rows = new Array(9).fill(0).map(() => Array(9).fill(0));
    const cols = new Array(9).fill(0).map(() => Array(9).fill(0));
    const boxes = new Array(9).fill(0).map(() => Array(9).fill(0));

    for (let i = 0; i < 9 ; i++)
    {
        for (let j = 0; j<9; j++)
        {
            if (board[i][j] !== '.')
            {
                const num = parseInt(board[i][j])
                const boxId = Math.floor(i / 3 ) *3 + Math.floor(j / 3)
                if (rows[i][num] || cols[j][num] || boxes[boxId][num])
                    return false;
                rows[i][num] = 1;
                cols[j][num] = 1;
                boxes[boxId][num] =1 ;
            }
        }
    }
    return true

}
